flocker/apiclient/_client.py

```diff
@@ -536,3 +556,17 @@ def to_nodes(result):
         request.addCallback(to_nodes)
 
         return request
+
+    def this_node_uuid(self):
+        era = check_output(["flocker-node-era"])
+        request = self._request(
+            b"GET", b"/state/nodes/by_era/" + era, None, {OK},
+            {NOT_FOUND: NotFound},
+        )
+
+        def handle_error(failure):
+            failure.trap(NotFound)
+            return deferLater(self._reactor, 0.1, self.this_node_uuid)
+        request.addCallbacks(lambda result: UUID(result["uuid"]), handle_error)
```

Comment:
Please use `flocker.common.retry_failure` here, e.g.

```python
retry_failure(self._reactor, lambda: self._request(...), [NotFound], repeat(0.1))
```

Also, is it correct to retry calls to `flocker-node-era` if it's not found on the API?